oak/server/rust/oak_runtime/src/node/wasm.rs

```diff
@@ -746,67 +746,143 @@ impl wasmi::ModuleImportResolver for WasmInterface {
     }
 }
 
-/// Create a new instance of a Wasm node. If the entry point is not found,
-/// `ERR(OakStatus::ErrInvalidArgs)` will be returned immediately.
-pub fn new_instance(
-    config_name: &str,
+pub struct WasmNode {
+    config_name: String,
     runtime: RuntimeRef,
     module: Arc<wasmi::Module>,
     entrypoint: String,
-    initial_reader: Handle,
-) -> Result<JoinHandle<()>, OakStatus> {
-    let config_name = config_name.to_owned();
-
-    debug!("new_instance: discovering {} {}", config_name, entrypoint);
-
-    // wasmi can't enumerate exports at creation, so we have to do it here per instance spawned as
-    // the entrypoint could be anything. We do it before spawning the child thread so that we can
-    // return an error code immediately if appropriate.
-    {
-        let (abi, _) = WasmInterface::new(config_name.clone(), runtime.clone(), initial_reader);
-        let instance = wasmi::ModuleInstance::new(
-            &module,
-            &wasmi::ImportsBuilder::new().with_resolver("oak", &abi),
-        )
-        .expect("failed to instantiate wasm module")
-        .assert_no_start();
-
-        let entrypoint_sig = wasmi::Signature::new(&[ABI_U64][..], None);
-
-        if instance
-            .export_by_name(&entrypoint)
-            .and_then(|e| e.as_func().map(|f| f.signature() != &entrypoint_sig))
-            .unwrap_or(true)
-        {
-            return Err(OakStatus::ErrInvalidArgs);
+    reader: Handle,
+    thread_handle: Option<JoinHandle<()>>,
+}
+
+impl WasmNode {
+    /// Creates a new [`WasmNode`] instance, but does not start it.
+    pub fn create(
```

Comment:
I think it also looks line `new` function?

Reply:
Done.